src/tools/miri/src/shims/time.rs

```diff
@@ -362,6 +362,71 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         interp_ok(Scalar::from_i32(0))
     }
 
+    fn clock_nanosleep(
+        &mut self,
+        clock_id: &OpTy<'tcx>,
+        flags: &OpTy<'tcx>,
+        req_op: &OpTy<'tcx>,
+        _rem: &OpTy<'tcx>, // Signal handlers are not supported, so rem will never be written to.
+    ) -> InterpResult<'tcx, Scalar> {
+        let this = self.eval_context_mut();
+
+        let clock_id: libc::clockid_t = this.read_scalar(clock_id)?.to_i32()?;
+        match clock_id {
+            libc::CLOCK_MONOTONIC => (),
+            libc::CLOCK_REALTIME
+            | libc::CLOCK_TAI
+            | libc::CLOCK_BOOTTIME
+            | libc::CLOCK_PROCESS_CPUTIME_ID => {
+                // The standard lib through sleep_until only needs CLOCK_MONOTONIC
+                panic!("MIRI only supports CLOCK_MONOTONIC for clock_nanosleep")
+            }
+            _other => return this.set_last_error_and_return_i32(LibcError("EINVAL")),
+        }
```

Comment:
This compares the *target* `click_id` value against the *host* constants, which makes no sense. Also, no need to even match on `CLOCK_TAI` etc if we don't support it.

The correct way to do this check is
```rust
if clock_id != this.eval_libc("CLOCK_MONOTONIC").to_int(clockid_t_size)? {
    throw_unsup_format!("clock_nanosleep: only CLOCK_MONOTONIC is supported");
}
```